lms/djangoapps/program_enrollments/api/tests/test_reading.py

```diff
@@ -443,6 +444,42 @@ def _usernames_to_users(dictionary):
             del result['usernames']
         return result
 
+    @ddt.data(
+        (
+            {'username': username_0, 'course_key': course_key_p},
+            None
+        ),
+        (
+            {'username': username_1, 'course_key': course_key_p},
+            None
+        ),
+        (
+            {'username': username_1, 'course_key': course_key_r},
+            None
+        ),
+        (
+            {'username': username_2, 'course_key': course_key_p},
+            None
+        ),
+        (
+            {'username': username_3, 'course_key': course_key_p},
+            ext_3
+        ),
+        (
+            {'username': username_3, 'course_key': course_key_r},
+            None
+        ),
+        (
+            {'username': username_4, 'course_key': course_key_p},
+            None
+        )
+    )
+    @ddt.unpack
+    def test_get_external_key_by_user_and_course(self, kwargs, expected_external_user_key):
```

Comment:
style nit: you could fit each case on a single line if the arguments to this function were `(self, username, course_key, expected_external_user_key)`

Reply:
I want to take advantage of the `self._username_to_user` function. If I were to expand the function signature, that function call would require the reconstruction of the dictionary.